File: java-samples/src/main/java/posts/parthmistry/javasamples/LogLineRegexCount.java
package posts.parthmistry.javasamples;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.regex.Pattern;

public class LogLineRegexCount {

    public static void main(String[] args) throws Exception {
        var logFilePath = args[0];

        var logLineCount = 0L;

        var pattern = Pattern.compile("[\\d]+-[\\d]+-[\\d]+ [\\d]+:[\\d]+:[\\d]+\\.[\\d]+ .* REQUEST on (.*)");

        try (var reader = Files.newBufferedReader(Paths.get(logFilePath))) {
            String line = null;
            while ((line = reader.readLine()) != null) {
                var matcher = pattern.matcher(line);
                if (matcher.matches()) {
                    logLineCount += 1;
                }
            }
        }

        System.out.println("matching log line count: " + logLineCount);
    }

}

File: python-samples/log_line_regex_count.py
import re
import sys

log_file_path = sys.argv[1]

log_line_count = 0

pattern = re.compile("[\\d]+-[\\d]+-[\\d]+ [\\d]+:[\\d]+:[\\d]+\.[\\d]+ .* REQUEST on (.*)")

with open(log_file_path, 'r') as log_file:
    while line := log_file.readline():
        processed_line = line.rstrip('\n')
        result = pattern.search(processed_line)
        if result is not None:
            log_line_count += 1

print("matching log line count:", log_line_count)
